### PR TITLE
Add Asp.net extensibility point for document highlighting.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/61862

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dibarbet this is ready for review.

File: src/Tools/ExternalAccess/AspNetCore/EmbeddedLanguages/DocumentHighlighting/IEmbeddedLanguageDocumentHighlighter.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    internal enum AspNetCoreHighlightSpanKind
+    {
+        None,
+        Definition,
+        Reference,
+        WrittenReference,
+    }
+
+    [DataContract]
```
Comment (CyrusNajmabadi) [inline src/Tools/ExternalAccess/AspNetCore/EmbeddedLanguages/DocumentHighlighting/IEmbeddedLanguageDocumentHighlighter.cs:32]: doesn't need to be a DC.  but i copy/pasted this from the internal defs

File: src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageDocumentHighlighter.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        public ImmutableArray<DocumentHighlights> GetDocumentHighlights(
+            Document document,
+            SemanticModel semanticModel,
+            SyntaxToken token,
+            int position,
+            HighlightingOptions options,
+            CancellationToken cancellationToken)
+        {
+            var highlighters = AspNetCoreDocumentHighlighterExtensionProvider.GetExtensions(document.Project);
+            foreach (var highlighter in highlighters)
```
Comment (dibarbet) [inline src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageDocumentHighlighter.cs:38]: do we ever expect there to be more than one aspnet implementation for each of the embedded language features?

**Review by dibarbet — APPROVED** [review]

File: src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageDocumentHighlighter.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageDocumentHighlighter.cs:38]: probably.... not.  but this just follows the same pattern for *everything* asp.net.... so just nicer to keep consistent :D

**Pull request merged**